cmd/govim/oswatcher_darwin.go

```diff
@@ -0,0 +1,63 @@
+// +build darwin
+
+package main
+
+import (
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/fsnotify/fsevents"
+)
+
+const fRemoved = fsevents.ItemRemoved | fsevents.ItemRenamed
+const fChanged = fsevents.ItemCreated | fsevents.ItemModified | fsevents.ItemChangeOwner
+
+func newOSWatcher(gomodpath string) (*osWatcher, error) {
+	dirpath := filepath.Dir(gomodpath)
+	dev, err := fsevents.DeviceForPath(dirpath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve device for path %v: %v", dirpath, err)
+	}
+
+	es := &fsevents.EventStream{
+		Paths:   []string{dirpath},
+		Latency: 200 * time.Millisecond,
+		Device:  dev,
+		Flags:   fsevents.FileEvents | fsevents.WatchRoot,
+	}
+
+	es.Start()
+
+	eventCh := make(chan fsEvent)
+	go func() {
+		for {
+			events, ok := <-es.Events
+			if !ok {
+				break
+			}
+			for i := range events {
+				event := events[i]
+				path := filepath.Join("/", event.Path)
```

Comment:
What's going on here?

Reply:
Oh, sloppy leftover. It should be the actual mount point, because the event paths are relative to the mount point. It uses https://developer.apple.com/documentation/coreservices/1447341-fseventstreamcreaterelativetodev underneath.

I'll add something that resolves the mount root and use that instead.